smdebug/pytorch/hook.py

```diff
@@ -263,9 +263,15 @@ def _collect_torch_profiling_data_if_profiler_enabled(self):
             return
         records = torch.autograd._disable_profiler()
         self.autograd_profiler_enabled = False
-        function_events = torch.autograd.profiler.EventList(
-            torch.autograd.profiler.parse_cpu_trace(records), use_cuda=self.use_cuda
-        )
+        if version.parse(torch.__version__) >= version.parse("1.7"):
```

Comment:
Can we move the helper functions `is_pt_1_6`, `is_pt_1_7`, etc from [this test utils file](https://github.com/awslabs/sagemaker-debugger/blob/master/tests/profiler/pytorch/utils.py) to smdebug and use these helper functions throughout this hook file? We have a similar structure in KerasHook There are two benefits to doing this:
- This would reduce code redundancy, so we don't have to repeat this `version.parse` logic everywhere we need to determine the current PT version.
- The util functions define a version more strictly, i.e. `is_pt_1_7` is only `1.7.x`. The difference is that when new PT 1.8.x or future versions are released, this block wouldn't get executed and potentially break (we basically maintain the invariant that unsupported versions = no metrics from autograd and nothing should error).